test/cipher.go

```diff
@@ -21,25 +24,274 @@ func HashBench(b *testing.B, hash func() hash.Hash) {
 
 // Benchmark a stream cipher.
 func StreamCipherBench(b *testing.B, keylen int,
-			cipher func([]byte) cipher.Stream) {
+	cipher func([]byte) cipher.Stream) {
 	key := make([]byte, keylen)
 	b.SetBytes(1024 * 1024)
 	data := make([]byte, 1024)
 	for i := 0; i < b.N; i++ {
 		c := cipher(key)
 		for j := 0; j < 1024; j++ {
-			c.XORKeyStream(data,data)
+			c.XORKeyStream(data, data)
 		}
 	}
 }
 
 // Benchmark a block cipher operating in counter mode.
+/*
+XXX Broken
 func BlockCipherBench(b *testing.B, keylen int,
-			bcipher func([]byte) cipher.Block) {
+	bcipher func([]byte) cipher.Block) {
 	StreamCipherBench(b, keylen, func(key []byte) cipher.Stream {
 		bc := bcipher(key)
-		iv := make([]byte,bc.BlockSize())
-		return cipher.NewCTR(bc,iv)
+		iv := make([]byte, bc.BlockSize())
+		return cipher.NewCTR(bc, iv)
 	})
 }
+*/
+
+// Compares the bits between two arrays returning the fraction
+// of differences. If the two arrays are not of the same length
+// no comparison is made and a -1 is returned.
+func BitDiff(a, b []byte) float64 {
+	if len(a) != len(b) {
+		return -1
+	}
+
+	mask1 := byte(1)
```

Comment:
Oh geesh, you're working about unrolling and micro-optimizing a function that's only used in a *test suite*???